test/IconFontTester/IconFontTester/MainPage.xaml.cs

```diff
@@ -0,0 +1,104 @@
+using System.Collections.ObjectModel;
+using System.Text;
+using Newtonsoft.Json;
+using Uno.IconFontTester;
+using Uno.IconFontTester.Models;
+using Windows.System;
+
+namespace IconFontTester;
+
+public sealed partial class MainPage : Page
+{
+    private string _searchTerm;
+
+    public MainPage()
+    {
+        this.InitializeComponent();
+    }
+
+    protected override async void OnNavigatedTo(NavigationEventArgs e)
+    {
+        base.OnNavigatedTo(e);
+        Refresh();
+    }
+
+    public string SearchTerm
+    {
+        get => _searchTerm;
+        set
+        {
+            _searchTerm = value;
+            UpdateSearch();
+        }
+    }
+
+    private List<IconItem> _allGlyphs = new List<IconItem>();
+
+    public ObservableCollection<IconItem> FilteredGlyphs { get; } = new ObservableCollection<IconItem>();
+
+    public async void OpenSelectionFolder()
+    {
+        var folder = await ApplicationData.Current.LocalFolder.CreateFolderAsync("UnoIconTest", CreationCollisionOption.OpenIfExists);
+        await Launcher.LaunchFolderAsync(folder);
+    }
+
+    public async void Refresh()
+    {
+        _allGlyphs.Clear();
+#if NETFX_CORE
+            var folder = await ApplicationData.Current.LocalFolder.CreateFolderAsync("UnoIconTest", CreationCollisionOption.OpenIfExists);
+            var file = (await folder.TryGetItemAsync("selection.json")) as StorageFile;
+#else
+        StorageFile file;
+        try
+        {
+            file = await StorageFile.GetFileFromApplicationUriAsync(new Uri("ms-appx:///Assets/Fonts/selection.json"));
+        }
+        catch (Exception)
+        {
+            var dialog = new ContentDialog()
+            {
+                Title = "Error",
+                Content = "The selection.json file was not found in the Assets/Fonts/ folder. Add it, and click Refresh to try again.",
+                CloseButtonText = "Ok",
+                XamlRoot = XamlRoot
+            };
+            await dialog.ShowAsync();
+            return;
```

Comment:
@MartinZikmund I have this exception for Android for example when launching the app locally:

`System.InvalidOperationException: 'Trying to set position of the dialog before it is associated with a visual tree. This can happen if the dialog's XamlRoot was not set.'`